src/viam/sdk/robot/client.hpp

```diff
@@ -81,54 +130,50 @@ class RobotClient {
     ///
     /// Because the return type here is a `Resource`, the user will need to manually
     /// cast to the desired type.
-    std::shared_ptr<Resource> resource_by_name(const ResourceName& name);
+    std::shared_ptr<Resource> resource_by_name(const Name& name);
 
     template <typename T>
     /// @brief Lookup and return a `shared_ptr` to a resource of the requested type.
     /// @param name The ordinary name of the resource.
     /// @throws `std::runtime_error` if the requested resource doesn't exist or is the wrong type.
     /// @return a `shared_ptr` to the requested resource.
     std::shared_ptr<T> resource_by_name(std::string name) {
-        ResourceName r;
         API api = API::get<T>();
-        *r.mutable_namespace_() = api.type_namespace();
-        *r.mutable_type() = api.resource_type();
-        *r.mutable_subtype() = api.resource_subtype();
-        *r.mutable_name() = std::move(name);
+        Name r = Name(api, "", name);
```

Comment:
(minor) I _think_ we still want to update this to `Name r(api, "", name);`. I'm a little fuzzy on some of these details but I believe what we have now uses the copy constructor (i.e., we're making a `Name`, and then copying it and assigning `r` to equal that copy), where as `Name r(api, "", name)` only constructs one `Name`. I could be wrong given that we're dealing with an rvalue `Name` though! Happy for @acmorrow to come in and correct me on this one :)

Reply:
You could just write it `Name r{api, "", name}`, but I don't think you even need `r` anymore. I'd expect the following to work fine:

```
auto resource = this->resource_by_name({API::get<T>(), "", std::move(name)});
```